vector_search/utils.py

```diff
@@ -199,25 +206,121 @@ def _chunk_documents(encoder, texts, metadatas):
     return recursive_splitter.create_documents(texts=texts, metadatas=metadatas)
 
 
+def _embedding_context(document, resource_type):
+    """
+    Get the embedding context for a document
+    based on the resource_type (learning resource vs content file)
+    """
+    if resource_type != CONTENT_FILE_TYPE:
+        return (
+            f"{document.get('title')} "
+            f"{document.get('description')} {document.get('full_description')}"
+        )
+    else:
+        return document.get("content", "")
+
+
 def _process_resource_embeddings(serialized_resources):
     docs = []
     metadata = []
     ids = []
     encoder = dense_encoder()
     vector_name = encoder.model_short_name()
     for doc in serialized_resources:
+        if not should_generate_embeddings(doc, COURSE_TYPE):
+            update_payload(doc, resource_type=COURSE_TYPE)
+            continue
         vector_point_key = doc["readable_id"]
         metadata.append(doc)
         ids.append(vector_point_id(vector_point_key))
-        docs.append(
-            f"{doc.get('title')} {doc.get('description')} {doc.get('full_description')}"
-        )
+        docs.append(_embedding_context(doc, COURSE_TYPE))
     if len(docs) > 0:
         embeddings = encoder.embed_documents(docs)
         return points_generator(ids, metadata, embeddings, vector_name)
     return None
 
 
+def update_payload(
+    serialized_document,
+    resource_type,
+):
+    """
+    Update the payload of a document in Qdrant
+    based on the resource_type (learning resource vs content file)
+    """
+    client = qdrant_client()
+    if resource_type != CONTENT_FILE_TYPE:
+        collection_name = RESOURCES_COLLECTION_NAME
+        QDRANT_PARAM_MAP = QDRANT_RESOURCE_PARAM_MAP
+        points = [vector_point_id(serialized_document["readable_id"])]
+    else:
+        collection_name = CONTENT_FILES_COLLECTION_NAME
+        QDRANT_PARAM_MAP = QDRANT_CONTENT_FILE_PARAM_MAP
+        params = {
+            "resource_readable_id": serialized_document["resource_readable_id"],
+            "key": serialized_document["key"],
+            "run_readable_id": serialized_document["run_readable_id"],
+        }
+        points = [
+            point.id
+            for point in retrieve_points_matching_params(
+                params,
+                collection_name=collection_name,
+            )
+        ]
+    payload = {}
+    for key in QDRANT_PARAM_MAP:
+        if key in serialized_document:
+            payload[QDRANT_PARAM_MAP[key]] = serialized_document[key]
+    if not all([points, payload]):
+        return
+    client.set_payload(
+        collection_name=collection_name,
+        payload=payload,
+        points=points,
+    )
+
+
+def should_generate_embeddings(serialized_document, resource_type):
```

Comment:
I think it makes sense to have seperate content file and learning resource verisons of this function instead of all the if statements. 
You can consider separate content file or resource specific versions of  update_payload(), and  _embedding_context() too

Reply:
good idea. I updated the code so they each have their own _embedding_context, update_payload and should_generate_embeddings methods